### PR TITLE
apps: make use of OSSL_STORE for generalized cert and CRLs loading

### PR DESCRIPTION
**Pull request by DDvO**
This generalizes the `load_certs()`, `load_cert()`, and `load_crls()` functions used by the CLI apps `ca`, `cmp`, `cms`, `ocsp`,
 `pkcs12`, `s_client`, `s_server`, `smime`, `verify`, and `x509` taking advantage of the OSSL_STORE mechanism.
This allows loading not only in PEM format but also in DER and PKCS#12 format and loading also from URIs.
It also enables using passwords for certificate input files, using a new `-passcerts` option of `pkcs12`
and, as far as it makes sense, any existing password input options for other apps.

This supersedes the core of my old PR #4930.
This is also strongly related to #12643.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: apps/include/apps.h [hunk 1]
```diff
@@ -105,7 +105,7 @@ X509_REQ *load_csr(const char *file, int format, const char *desc);
 X509 *load_cert_pass(const char *uri, int maybe_stdin,
                      const char *pass, const char *desc);
```
Comment (levitte) [inline apps/include/apps.h:106]: Hmmmm, are the certs ever passphrase protected?  Yes, I understand that the *keys* in a PKCS#12 object may be protected, but the certs...?  Mind you, I'm still confused by PKCS#12, so I'm ready for a surprise :wink:  However, if the certs can be extracted without a passphrase, it's possible that the the 'file:' loader needs a bit of tuning...

Comment (t8m) [reply apps/include/apps.h:106]: I suppose the certs might not be passphrase protected in some special cases but usually IMO they are encrypted as well (although with stupidly weak encryption).

Comment (levitte) [reply apps/include/apps.h:106]: Ah, it's really about the whole PKCS#12 file, not the included certificates...  of course.  Yeah ok.

**Review by levitte — APPROVED** [review]

**Comment by levitte** [comment]
I think this finalizes a lot of what I did in #7390.  When this is merged, I'll revisit that one to see what remains.

**Comment by DDvO** [comment]
> I think this finalizes a lot of what I did in #7390. When this is merged, I'll revisit that one to see what remains.

Ah, I was not aware of that PR!

At times I think it would be good to have a mechanism that points out overlaps between existing PRs,
or even better, that could be used to detect if a planned change could overlap with existing PRs.

**Comment by levitte** [comment]
> Ah, I was not aware of that PR!

No matter, I'm actually glad to see someone else doing this, that indicates the desire for this change.

**Comment by levitte** [comment]
This is ready to be merged

**Comment by DDvO** [comment]
> This is ready to be merged

Done - thanks @levitte

**Comment by kaduk** [comment]
> At times I think it would be good to have a mechanism that points out overlaps between existing PRs,
or even better, that could be used to detect if a planned change could overlap with existing PRs.

gerrit code review will link to changes that conflict or are "related to" the change in question.  (I am not 100% sure of the details of what determines those relationships, and it might be configurable, but I think that the conflicts are actual 'git merge' failures and the "related"ness is parent/child relationships.

**Comment by DDvO** [comment]
> > At times I think it would be good to have a mechanism that points out overlaps between existing PRs,
> > or even better, that could be used to detect if a planned change could overlap with existing PRs.
> 
> gerrit code review will link to changes that conflict or are "related to" the change in question.

This sounds interesting!
Are there plans to integrate this with GitHub or OpenSSL?

**Pull request merged**